src/app/course-list/course-list.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+import { ICourse } from '../common/course.interface';
+import { MESSAGES } from '../common/constants/course.constants';
+
+import { CourseService } from '../common/services/course.service';
+
+@Component({
+  selector: 'app-course-list',
+  templateUrl: './course-list.component.html',
+  styleUrls: ['./course-list.component.scss']
+})
+export class CourseListComponent implements OnInit {
+
+  @Input() courses: ICourse[];
+
+  public noDataMessage = MESSAGES.NO_DATA_MESSAGE;
+
+  constructor(private courseService: CourseService) { }
+
+  ngOnInit() {}
+
+  deleteCourse(courseId: string) {
+    this.courseService
+      .deleteCourse(this.courses, courseId)
+      .subscribe(res => {
+        if (res) {
```

Comment:
Condition should be in filter operator

Reply:
Better will be to refactor CourseSercice in that way where the course list will be stored inside the Subject.
CourseSercice can looks like:
```tsscript
export class CourseSercice {
 protected courses: BehaviorSubject<ICourse[]> = new BehaviorSubject([]); 

 constructor() {}

 getCourses(): Observable<ICourse[]> {
  return this._courses.asObservable();
 }

 setCourses(courses: ICourse[]) {
  this.courses.next(courses);
 }

 deleteCourse(id: string) {
   this.getCourses().pipe(
    take(1)
   ).subscribe(
     courses => this.setCourses(courses.filter(
       course => course.id !== id
     ))
   )
 }
}
```
and at component

```tsscript
export class CoursePageComponent implements OnInit{
  

  constructor(private courseService: CourseSercice ) {}

  ngOnInit() {
    this.courseService.setCourses(COURSES);
  }

 deletCourse(id: string) {
   this.courseService.deleteCourse(id);
 }
}
```

Of course, the best would be to create `CourseDataSource` and `CourseService` to separate concerns.

Reply:
The best way will be create store and manage all operations with actions, reducers and effects.

Reply:
ngrx store will be implemented in upcoming homework. For now, this implementation is enough